tests/test_bucket_manifest.py

```diff
@@ -0,0 +1,117 @@
+import pytest
+from unittest.mock import MagicMock
+from contextlib import contextmanager
+
+import boto3
+from botocore.stub import Stubber
+from botocore.exceptions import ClientError
+
+from batch_jobs.bucket_manifest.object_metadata_job import (
+    compute_object_metadata,
+    send_message,
+)
+from batch_jobs.bucket_manifest.bucket_manifest_job import (
+    get_messages_from_queue,
+    submit_job,
+    list_objects,
+)
+
+
+@contextmanager
+def mock_file(filepath, content=""):
+    with open(filepath, "w") as f:
+        f.write(content)
+    yield filepath
+    try:
+        os.remove(filepath)
+    except Exception:
+        pass
+
+
+def test_compute_object_metadata_success(monkeypatch, s3):
+
+    monkeypatch.setattr(
+        "batch_jobs.bucket_manifest.object_metadata_job.BUCKET", "test_bucket"
+    )
+    monkeypatch.setattr(
+        "batch_jobs.bucket_manifest.object_metadata_job.S3KEY", "test_key"
+    )
+
+    monkeypatch.setattr("batch_jobs.bucket_manifest.object_metadata_job.MAX_RETRIES", 1)
+    output = compute_object_metadata()
+    assert output == {
+        "url": "s3://test_bucket/test_key",
+        "md5": "d9673f3128fcfbd70d040f7dc18afbd8",
+        "size": 7,
+    }
+
+
+def test_compute_object_metadata_fail_due_to_wrong_key(monkeypatch, s3):
+
+    monkeypatch.setattr(
+        "batch_jobs.bucket_manifest.object_metadata_job.BUCKET", "test_bucket"
+    )
+    monkeypatch.setattr(
+        "batch_jobs.bucket_manifest.object_metadata_job.S3KEY", "test_key2"
+    )
+
+    monkeypatch.setattr("batch_jobs.bucket_manifest.object_metadata_job.MAX_RETRIES", 1)
+    output = compute_object_metadata()
+    assert output == {
+        "url": "s3://test_bucket/test_key2",
+        "ERROR": "An error occurred (NoSuchKey) when calling the GetObject operation: The specified key does not exist.",
+    }
+
+
+def test_send_message_success(create_mock_sqs):
+    assert send_message("test", {})
+
+
+def test_send_message_fail_due_to_queue_not_exit(create_mock_sqs):
+    with pytest.raises(ClientError):
+        send_message("test2", {})
```

Comment:
where is the logic that forces this send_message to raise a clienterror?

Reply:
If the queue does not exist, it will raise a client error
https://github.com/uc-cdis/aws-batch-jobs/blob/1c5a6aabd1e888edbfc23fc12e1955d1475d0908/batch_jobs/bucket_manifest/object_metadata_job.py#L108

Reply:
ah, gotcha :+1: